### PR TITLE
ADD: code of conduct to repo and update contributing guide with current workflow

### LINKED ISSUE #274: Add code of conduct and contributing guide to stavalib
Github and other package health metrics always suggest having a clear

contributing guide and 
code of conduct 

to ensure that community development around a package is and remains healthy.

[See SKYK package health community metrics at the bottom:](https://snyk.io/advisor/python/stravalib)

and [Github community standards](https://github.com/stravalib/stravalib/community)

We should add a code of conduct to this repo and a better contributing guide.

### PR DESCRIPTION
**Pull request by lwasser**
closes #274, #269

Please see the open issue #274 for explanation. 
This pr:

- [x] Adds badges to the readme addressing #269 
- [x] Adds a code of conduct and uses the template from the open source contributors code of conduct 
- [x] It also creates an updated contributing guide.

In a separate PR i'll submit a new development guide (the file linked to in this pr). i'm just trying to keep my pr's a bit smaller as that pr was going to be 15 files with all of the changes!

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Change Log
 
 ## Unreleased
+* Add: add code of conduct to the repo and update contributing guide (@lwasser, #274) 
 * Fix: deprecated set-output command in actions build (@yihong0618, #272)
```
Comment (lwasser) [inline changelog.md:5]:
```suggestion
* Add: add code of conduct to the repo, update contributing guide + readme badges (@lwasser, #269, #274) 
* Fix: deprecated set-output command in actions build (@yihong0618, #272)
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
+# How to Contribute to Stravalib
+
+**Attention:** This contributing guide provides an high-level overview for contributing
```
Comment (jsamoocha) [inline CONTRIBUTING.md:3]: a~~n~~ high-level overview ?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+If you are thinking about submitting a change to stravalib, please start
+by [submitting an issue in our GitHub repository](https://www.github.com/stravalib/stravalib/issues/). 
+We will use that issue to communicate with you about
+
+1. Whether the change is in scope for the project
+2. Any issues that you need help with or clarification on. 
+
+We welcome changes to both documentation and code. 
+
+### Want to submit a pull request with changes to our code of documentation?
```
Comment (jsamoocha) [inline CONTRIBUTING.md:38]: code ~~of~~ or documentation

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+
+We welcome changes to stravalib through pull requests. Before you submit a 
+pull request please sure to :
```
Comment (jsamoocha) [inline CONTRIBUTING.md:41]: please [make/be?] sure to

**Review by jsamoocha — APPROVED** [review]
Thanks, this is a great addition! It can be merged once the few minor typos pointed to in the comments are resolved.

File: CONTRIBUTING.md (hunk 2 above)
Comment (lwasser) [reply CONTRIBUTING.md:3]: fixing now! thanks for reviewing and catching these typos!!!

File: CONTRIBUTING.md (hunk 3 above)
Comment (lwasser) [reply CONTRIBUTING.md:38]: fixed!

File: CONTRIBUTING.md (hunk 4 above)
Comment (lwasser) [reply CONTRIBUTING.md:41]: fixed!!

**Comment by lwasser** [comment]
ok great.  i'ved fixed the items that you caught! Two more documentation PR's are coming. i am trying to split them up to avoid another massive PR. :) Thanks for the review @jsamoocha !!

**Comment by lwasser** [comment]
ok all checks building and typos fixed. I am going to merge, rebase and submit the next set of documentation updates  on top of this set :)

**Pull request merged**